operator/src/config.rs

```diff
@@ -125,14 +156,122 @@ pub fn create_volumes(configmap_name: &str) -> Vec<Volume> {
 pub fn create_required_startup_env() -> Vec<EnvVar> {
     vec![
         EnvVar {
-            name: "SPARK_NO_DAEMONIZE".to_string(),
+            name: SPARK_NO_DAEMONIZE.to_string(),
             value: Some("true".to_string()),
             ..EnvVar::default()
         },
         EnvVar {
-            name: "SPARK_CONF_DIR".to_string(),
+            name: SPARK_CONF_DIR.to_string(),
             value: Some("{{configroot}}/conf".to_string()),
             ..EnvVar::default()
         },
     ]
 }
+
+/// Get all required configuration options
+/// 1) from spec
+/// 2) from node
+/// 3) from selector
+/// 4) from config properties
+/// # Arguments
+/// * `spec` - SparkCluster spec for common properties
+/// * `selector` - SparkClusterSelector containing desired config properties
+///
+pub fn get_config_properties(
```

Comment:
This seems to mix configuration for different types of nodes into one?
History Server & Worker stuff etc.?
Fine for now but in the future it'd be good if the configs are as minimal as possible.

Reply:
With the hashed selector approach we only create one config map per selector (which may be shared for multiple master/worker...). That means e.g. a selector for a master wont have any properties from a history server.